pkg/envinfo/envinfo.go

```diff
@@ -170,6 +170,18 @@ func newProxyEnvInfo() proxyEnvInfo {
 func (esi *EnvSpecificInfo) SetConfiguration(parameter string, value interface{}) (err error) {
 	if parameter, ok := asLocallySupportedParameter(parameter); ok {
 		switch parameter {
+		case "name":
+			val := value.(string)
+			esi.componentSettings.Name = val
+		case "namespace":
+			val := value.(string)
+			esi.componentSettings.Namespace = val
+		case "debugport":
+			val, err := strconv.Atoi(value.(string))
+			if err != nil {
+				return errors.Wrap(err, "failed to set debug port")
+			}
+			esi.componentSettings.DebugPort = &val
 		case "url":
```

Comment:
I get a panic when I try to add a url with `odo env`:
```
Johns-MacBook-Pro-3:tester johncollier$ odo env set url test.com
panic: interface conversion: interface {} is string, not envinfo.EnvInfoURL

goroutine 1 [running]:
github.com/openshift/odo/pkg/envinfo.(*EnvSpecificInfo).SetConfiguration(0xc0007f32c0, 0x7ffeefbffa0c, 0x3, 0x25289c0, 0xc0008afc60, 0x2325e73, 0x0)
	/Users/johncollier/go/src/github.com/openshift/odo/pkg/envinfo/envinfo.go:186 +0x8a6
github.com/openshift/odo/pkg/odo/cli/env.(*SetOptions).Run(0xc000153e80, 0x0, 0x0)
	/Users/johncollier/go/src/github.com/openshift/odo/pkg/odo/cli/env/set.go:80 +0x94
github.com/openshift/odo/pkg/odo/genericclioptions.GenericRun(0x2b37560, 0xc000153e80, 0xc00056ca00, 0xc000844840, 0x2, 0x2)
	/Users/johncollier/go/src/github.com/openshift/odo/pkg/odo/genericclioptions/runnable.go:31 +0x13c
github.com/openshift/odo/pkg/odo/cli/env.NewCmdSet.func2(0xc00056ca00, 0xc000844840, 0x2, 0x2)
	/Users/johncollier/go/src/github.com/openshift/odo/pkg/odo/cli/env/set.go:108 +0x5e
github.com/openshift/odo/vendor/github.com/spf13/cobra.(*Command).execute(0xc00056ca00, 0xc000844800, 0x2, 0x2, 0xc00056ca00, 0xc000844800)
	/Users/johncollier/go/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:830 +0x29d
github.com/openshift/odo/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc000256c80, 0x292d270, 0xc0005cf320, 0x4)
	/Users/johncollier/go/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:914 +0x2fb
github.com/openshift/odo/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/Users/johncollier/go/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:864
main.main()
	/Users/johncollier/go/src/github.com/openshift/odo/cmd/odo/odo.go:67 +0x32a
```

Reply:
Actually the `odo env` doesn't support url set/unset because `odo url` already covers the implementation of url configuration. We discussed this before and more design details can be found here: https://github.com/openshift/odo/issues/3401#issuecomment-659623708

I can add more validation code to avoid the panic issue of setting unsupported parameters. Also I can make the usage more clear to let user know what parameters are currently supported by `odo env` by running `odo env --help`.

Reply:
Then we shouldn’t allow setting it, we should error out

Reply:
@girishramnani Yes, that's the plan, the validation code that I mention above will handle that.

Reply:
It does the same for `odo env set push <value>`, so make sure that one is handled too